### PR TITLE
TQL syntax highlighting in docs site

### PR DESCRIPTION
**Pull request by dit7ya**
This PR adds support for syntax highlighting of TQL code blocks in the docs site.

To review, add a TQL code block in any docs or blog page, and run docusaurus.

Missing feature: copy to clipboard button, see upstream issue - https://github.com/rehype-pretty/rehype-pretty-code/issues/235.

Closes https://github.com/tenzir/issues/issues/2035.

### CODE REVIEW

File: web/package.json
Comment (jachris) [inline web/package.json:1]: How can I try this out? I get a 404:

```
$ yarn
yarn install v1.22.22
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error Error: https://pkg.pr.new/tenzir/vscode-tql/tql@3: Request failed "404 Not Found"
    at ResponseError.ExtendableBuiltin (/opt/homebrew/Cellar/yarn/1.22.22/libexec/lib/cli.js:696:66)
    at new ResponseError (/opt/homebrew/Cellar/yarn/1.22.22/libexec/lib/cli.js:802:124)
    at Request.<anonymous> (/opt/homebrew/Cellar/yarn/1.22.22/libexec/lib/cli.js:66750:16)
    at Request.emit (node:events:519:28)
    at module.exports.Request.onRequestResponse (/opt/homebrew/Cellar/yarn/1.22.22/libexec/lib/cli.js:142287:10)
    at ClientRequest.emit (node:events:519:28)
    at HTTPParser.parserOnIncomingClient (node:_http_client:698:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
    at TLSSocket.socketOnData (node:_http_client:540:22)
    at TLSSocket.emit (node:events:519:28)
```

Comment (dit7ya) [reply web/package.json:1]: This seems strange. Apparently the build succeeded locally and also on CI, so I am wondering what might be the case?!

Comment (jachris) [reply web/package.json:1]: JS package delivery is not really my specialty. But maybe it's somehow guarded by permissions? After all, the associated repository is also private. I think we want the docs to build also for people without any special permissions.

Comment (dit7ya) [reply web/package.json:1]: We have published vscode-tql on npm now.

File: web/docusaurus.config.js [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
 
+const tqlGrammar = require("@tenzir/vscode-tql/syntaxes/tql.tmLanguage.json");
+
 async function createConfig() {
+  const rehypePrettyCode = (await import('rehype-pretty-code')).default;
+  const shikiHighlighter = (await import('shiki')).createHighlighter;
+
+  const rehypePrettyCodeOptions = {
+    keepBackground: false,
+    theme: 'github-dark-dimmed',
```
Comment (jachris) [inline web/docusaurus.config.js:14]:
```suggestion
    theme: 'github-dark-default',
```

The contrast of `-dimmed` is very poor.

Comment (jachris) [inline web/docusaurus.config.js:14]: We could also change the background color to something more dark.

**Review by jachris — COMMENTED** [review]
Thanks, this seems to work now!

File: web/docusaurus.config.js (hunk 1 above)
Comment (dominiklohmann) [reply web/docusaurus.config.js:14]: Somewhat related, and I also already chatted with @dit7ya about this: We should try to get the light theme up and running for light mode docs, as that's what we embed in the Tenzir Platform as well.

Comment (dit7ya) [reply web/docusaurus.config.js]: Changed the theme, but getting two themes working seems to be very hard without upgrading MDX.

Comment (dominiklohmann) [reply web/docusaurus.config.js]: Okay, let's stick with the dark theme for now then and get back to that later. Can you write a follow-up story before merging this?

Comment (dit7ya) [reply web/docusaurus.config.js]: I added it to this existing story. 

https://github.com/tenzir/issues/issues/1132

**Review by jachris — APPROVED** [review]
Are there any remaining concerns here, or can we merge this?

I would personally wish for slightly more contrast between background and default color than what we currently have. Ignore that this example should use the TQL highlighter in the first place, it's just an example:

image

But it's already okay and I think we can address this later if need be.

**Comment by dominiklohmann** [comment]
Let's merge it and then iterate on it, @jachris. Better to get this in than to leave it open.

One follow-up we should resolve sooner rather than later is filling in missing tql tags:

~~~
rg --multiline '```\n\/\/ tql2'
~~~

**Pull request merged**